src/components/app.tsx

```diff
@@ -79,8 +82,8 @@ export class App extends React.PureComponent<IProps, IState> {
       this.setState(newState as IState);
 
       if (teacherEditionMode) {
-        createPluginNamespace();
-        loadPluginScripts(activity);
+        this.LARA = initializeLara();
+        loadPluginScripts(this.LARA, activity);
```

Comment:
One thing that is dawning on me - do we need to be careful about checking when `loadPluginScripts` completes?  Is there a chance that somewhere we might be trying to access a script that isn't loaded?

Reply:
That is a good point. In LARA the scripts are injected on the page by the server. So they are guaranteed to be loaded  before the code after them is called. 

However I think based on the plugin design it should be safe: 🤞 . The plugin should include a call to register itself which it wouldn't be able to do until its own code has been loaded. And before the plugin is registered there shouldn't be anything that is trying to access it.